### DISCUSSION TITLE
[testing] RHEL/RHEL-like/RHEL-clone and other distribution testing

### DISCUSSION THREAD

**Discussion by TurboTurtle** [sosreport/sos#3642] [General]
Forking this discussion off from #3635.

We've long desired to test directly on RHEL given that Red Hat is one of our two major sponsors. We get "pretty good" test coverage via CentOS and Fedora, but with the ongoing GCE image issues from Fedora that seems to be getting more limited as we go on (side note: if the Fedora project isn't going to fix pushing images to GCE, then we need to seriously talk about dropping it from our CI given the current versions we test on are either no longer supported or very soon to be unsupported).

The main issue is that by default launching a RHEL instance on cloud providers (not just GCE like we use) typically invokes a subscription charge. RH provides a method around this known, at least at one point, as "Cloud Access". The TL;DR is that when your project is configured for Cloud Access you are able to launch from a _different_ set of base images that come directly from RH and don't incur the subscription charge. When I left RH I was trying to get the sos GCE project setup for that, but due to a lot of internal bits flying the conversation never went very far.

The ideal situation is that the RH folks can pick that up and get our GCE project correctly configured and provide the image IDs we need to launch from. @pmoravec @bmr-cymru @mhradile @jcastill tagging for visibility.

Failing that, let's talk about what we can do otherwise. CentOS Stream is an upstream for RHEL these days, and there are two RHEL clones that are binary compatible rebuilds - Rocky and Alma - that, once #3635 is merged, we'll have direct support for in sos. Is it "good enough" to test on, e.g. Alma 9.2 to be more confident in our RHEL releases and getting more testing done in upstream? Do we want to add Alma testing *anyway* even if we get the Cloud Access situation figured out? Do we need/want some threshold established for adding a distribution to our testing CI? We've pushed back on arbitrarily adding policies of RHEL clones in the past due to the tendency of those distros to die out and not have any further involvement, so does it make sense to formally adopt a position that if we add a policy we'll add testing and thus to add a policy we want a certain level of engagement with the project?

**Comment by jcastill** [comment]
After discussing the situation internally with @pmoravec and @mhradile, we've decided that the CentOS Stream images are good enough for RHEL testing.
We are investigating possibilities regarding Fedora GCP images, but we don't have any news about this yet.

**Comment by arif-ali** [comment]
Considering the EOL of centos stream 8, is it worth doing testing on Alma, oracle or rocky 8? I believe these would be similar and they have a longer support cycle

**Comment by jcastill** [comment]
We still have some things to try and get Stream 8 images, so lets go through them first. If these options don't work for us, then we can evaluate alternatives, or drop RHEL8 completely.

**Reply by arif-ali** [reply]
sounds good, I had my own local scripts that does avocado on centos8, and now converted to alma/rocky and now added fedora38,39,40 to my tests and they all work as expected. thought it would be the quickest way to test things and move things along :)
